interactive_engine/executor/ir/runtime/src/process/entry.rs

```diff
@@ -0,0 +1,619 @@
+//
+//! Copyright 2022 Alibaba Group Holding Limited.
+//!
+//! Licensed under the Apache License, Version 2.0 (the "License");
+//! you may not use this file except in compliance with the License.
+//! You may obtain a copy of the License at
+//!
+//! http://www.apache.org/licenses/LICENSE-2.0
+//!
+//! Unless required by applicable law or agreed to in writing, software
+//! distributed under the License is distributed on an "AS IS" BASIS,
+//! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//! See the License for the specific language governing permissions and
+//! limitations under the License.
+
+use std::any::Any;
+use std::cmp::Ordering;
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use std::fmt::Debug;
+use std::hash::{Hash, Hasher};
+use std::sync::Arc;
+
+use dyn_type::{BorrowObject, Object};
+use graph_proxy::apis::{
+    read_id, write_id, DynDetails, Edge, Element, GraphElement, GraphPath, Vertex, ID,
+};
+use ir_common::error::ParsePbError;
+use ir_common::generated::results as result_pb;
+use pegasus::codec::{Decode, Encode, ReadExt, WriteExt};
+use pegasus_common::downcast::*;
+use pegasus_common::impl_as_any;
+
+use crate::process::operator::map::Intersection;
+
+#[derive(Debug)]
+pub enum EntryDataType {
+    Vid, // A vertex global id
```

Comment:
Do not assume `Vid`. `Id` is fine.

Reply:
Use `Vid` here since we may need to specify whether it is a `Vid` or `Eid` when getting properties in `Auxilia`. It would be modified as an `Id` once we separate `GetVProp` and `GetEProp` from `Auxilia`.